Userland/Libraries/LibGL/SoftwareGLContext.cpp

```diff
@@ -3495,4 +3593,125 @@ void SoftwareGLContext::gl_color_material(GLenum face, GLenum mode)
     m_light_state_is_dirty = true;
 }
 
+void SoftwareGLContext::gl_get_light(GLenum light, GLenum pname, void* params, GLenum type)
+{
+    APPEND_TO_CALL_LIST_AND_RETURN_IF_NEEDED(gl_get_light, light, pname, params, type);
+    RETURN_WITH_ERROR_IF(m_in_draw_state, GL_INVALID_OPERATION);
+    RETURN_WITH_ERROR_IF(light < GL_LIGHT0 || light > GL_LIGHT0 + m_device_info.num_lights, GL_INVALID_ENUM);
+    RETURN_WITH_ERROR_IF(!(pname == GL_AMBIENT || pname == GL_DIFFUSE || pname == GL_SPECULAR || pname == GL_SPOT_DIRECTION || pname == GL_SPOT_EXPONENT || pname == GL_SPOT_CUTOFF || pname == GL_CONSTANT_ATTENUATION || pname == GL_LINEAR_ATTENUATION || pname == GL_QUADRATIC_ATTENUATION), GL_INVALID_ENUM);
+
+    if (type == GL_FLOAT)
+        get_light_param<GLfloat>(light, pname, static_cast<GLfloat*>(params));
```

Comment:
You should be able to drop the `<GLfloat>` here I think? It's inferred by the third argument

Reply:
I don't think so. Because it's a `void` pointer, I believe we need to massage the compiler into emitting the right type, especially because we're using a single function as the entry point, but I'll page Dr. @alimpfard just to double check.

Reply:
Just checked, removing this causes a compilation problem.